### PR TITLE
feature: Async Locals

### PR DESCRIPTION
**Pull request by icirellik**
Leverages the node api `AsyncLocalStorage` to provide safe access to
shared context across async requests.

https://nodejs.org/api/async_hooks.html#async_hooks_class_asynclocalstorage

### CODE REVIEW

File: src/lib/asyncLocalWrapper/browser.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import type { AsyncLocalStorage } from "async_hooks"
+
+const map = new Map()
+
+// A no-op stub incase this is access in the browser.
```
Comment (damassi) [inline src/lib/asyncLocalWrapper/browser.ts:5]:
```suggestion
// A no-op stub in case this is accessed in the browser.
```

**Review by damassi — APPROVED** [review]
LGTM 👍

**Comment by mzikherman** [comment]
Let's merge and give this a shot?

**Comment by dzucconi** [comment]
Seeing the runtime error `TypeError: _async_hooks.AsyncLocalStorage is not a constructor`?

**Comment by damassi** [comment]
Might be worth bringing this to #product-platform

**Pull request merged**